abi/known.xml

```diff
@@ -1,52 +1,57 @@
-<interfaces>
-
-    <!--    Text    -->
-    <interface name="text_comment" group="text">
-        <internal name="text_comment">
-            <input>
-                text_comment#00000000 text:Text = InternalMsgBody;
-            </input>
-        </internal>
-    </interface>
-
-    <!--    Wallet V3R2    -->
-    <interface name="wallet_v3r2" group="wallet">
-        <get_method name="seqno">
+<abi>
+    <types interface="tep62">
+        snake#00 data:(SnakeData ~n) = ContentData;
+        chunks#01 data:ChunkedData = ContentData;
+        onchain#00 data:(HashmapE 256 ^ContentData) = FullContent;
+        offchain#01 uri:Text = FullContent;
+    </types>
+    <types interface="payment_channel">
+        cp#_ amount:Coins condition:Cell = ConditionalPayment;
+        sc_body#_ seqno:uint64 sent:Coins conditionals:(HashmapE 32 ConditionalPayment)
+        = SemiChannelBody; // 64+132+1 = 197
+        semichannel_state#43685374 channel_id:uint128 data:SemiChannelBody counterparty_data:(Maybe ^SemiChannelBody)
+        = SemiChannel; // 32 + 128 + 197 + 1 = 358
+        signed_schs#_ signature:bits512 state:SemiChannel = SignedSemiChannel; // 512 + 358 = 870
+        quarantined_state#_ state_A:SemiChannelBody state_B:SemiChannelBody
+        quarantine_starts:uint32
+        state_commited_by_A:Bool
+        = QuarantinedState; // 358 + 358 + 32 + 1 = 749
+        conf#_ quarantin_duration:uint32
+        misbehavior_fine:Coins
+        conditional_close_duration:uint32 = ClosingConfig; // 32 + 132 + 132 + 32 = 296
+        payment_conf#_  excess_fee:Coins
+        dest_A:MsgAddress
+        dest_B:MsgAddress = PaymentConfig;
+        channel_state#_ balance_A:Coins balance_B:Coins
+        key_A:uint256 key_B:uint256
+        channel_id:uint128
+        config:^ClosingConfig
+        commited_seqno_A:uint32 commited_seqno_B:uint32
+        quarantin:(Maybe ^QuarantinedState)
+        payments:^PaymentConfig = Storage;
+        // 132 + 132 + 256 + 256 + 32 + 32 + 1 + 132 = 973
+    </types>
+    <types interface="storage_provider">
+        torrent_info piece_size:uint32 file_size:uint64 root_hash:(## 256) header_size:uint64 header_hash:(## 256)
+        microchunk_hash:(Maybe (## 256)) description:Text = TorrentInfo;
+    </types>
+        <!--        Wallets      -->
+        <get_method name="seqno" interface="wallet" interface="storage_provider">
             <output>
                 <tinyint name="state">uint32</tinyint>
             </output>
         </get_method>
-        <get_method name="get_public_key">
+        <get_method name="get_public_key" interface="wallet" interface="storage_provider">
```

Comment:
I'm not sure that having two attributes with the same name is a good idea. 
probably, more widely used approaches are:
```xml
<get_method interface="wallet storage" />
```
or
```xml
<interfaces>
    <interface>storage</interface>
    <interface>wallet</interface>
</interfaces>
```

Reply:
>I'm not sure that having two attributes with the same name is a good idea.
golang std library parses it to slice without problem. but it can create problem in java.  i think we can easily can change it in the future if it creating problem.